### PR TITLE
Fix bug where the `fix_even_unparsable` setting was not being respected in `.sqlfluff`

### LINKED ISSUE #3214: Config for fix_even_unparsable not being applied
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When setting the any config file to `fix_even_unparsable = True` the config get's overriden by the default (or lack thereof) on the @click.option decorator for the fix command.

### Expected Behaviour

When setting the config `fix_even_unparsable` it should be captured by the fix command as well.

### Observed Behaviour

The `fix_even_unparsable` command is not being captured by the fix command

### How to reproduce

Create a config file and include `fix_even_unparsable`
Run `sqlfluff fix`
Note that `fix_even_unparsable` is set to False at runtime

### Dialect

Any

### Version

0.13.0

### Configuration

`pyproject.toml`

```
[tool.sqlfluff.core]
verbose = 2
dialect = "snowflake"
fix_even_unparsable = true
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by barrywhart**
### Brief summary of the change made
Fixes #3214 

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: test/cli/commands_test.py [hunk 1]
```diff
@@ -687,6 +687,55 @@ def test__cli__fix_error_handling_behavior(sql, fix_args, fixed, exit_code, tmpd
...
+        str(tmpdir / "fix_even_unparsable.sql"),
+    ]
+    if method == "command-line":
+        if fix_even_unparsable:
+            options.append("--FIX-EVEN-UNPARSABLE")
+    else:
+        assert method == "config-file"
+        with open(str(tmpdir / ".sqlfluff"), "w") as f:
+            print(f"[sqlfluff]\nfix_even_unparsable = {fix_even_unparsable}", file=f)
+    monkeypatch.chdir(str(tmpdir))
```
Comment (barrywhart) [inline test/cli/commands_test.py:719]: Without this line, the 4th test case fails.

File: test/cli/commands_test.py [hunk 2]
```diff
@@ -687,6 +687,59 @@ def test__cli__fix_error_handling_behavior(sql, fix_args, fixed, exit_code, tmpd
...
+            options,
+        ],
+    )
+    fixed_path = str(tmpdir / "fix_even_unparsableFIXED.sql")
+    if fix_even_unparsable:
+        with open(fixed_path, "r") as f:
+            fixed_sql = f.read()
+            assert (
+                fixed_sql
+                == "SELECT my_col\nFROM my_schema.my_table\nWHERE processdate ! 3\n"
```
Comment (tunetheweb) [inline test/cli/commands_test.py:737]: Why are we adding this as inline SQL for the output, but have a file for the input?

Could we do both as inline, or both as files?

**Review by tunetheweb — APPROVED** [review]
LGTM

One nit/suggestion

File: test/cli/commands_test.py (hunk 2 above)
Comment (barrywhart) [reply test/cli/commands_test.py:737]: I could do both as inline -- will change. I adapted this test from an existing one that used different file encodings, so in that case, files definitely made sense for the input.

**Review by tunetheweb — CHANGES_REQUESTED** [review]
Could you also fix this line as part of this PR?

https://github.com/sqlfluff/sqlfluff/blob/262010b91cf5616de242dad504c788e9cd33ac58/src/sqlfluff/cli/commands.py#L753

Should be `--FIX-EVEN-UNPARSABLE` in caps.

**Comment by barrywhart** [comment]
@tunetheweb: Yup, changing that now.

**Comment by barrywhart** [comment]
Done!

**Review by tunetheweb — APPROVED** [review]
Since we've fixed two of these in quick succession recently, are there any other command line options we should look at that need a similar fix?

**Comment by barrywhart** [comment]
I looked over the file `commands.py` and don't see any obvious issues like this.

**Pull request merged**